functions/snippets/src/main/java/com/example/functions/OcrTranslateText.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+// [START functions_ocr_translate]
+
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.translate.v3.LocationName;
+import com.google.cloud.translate.v3.TranslateTextRequest;
+import com.google.cloud.translate.v3.TranslateTextResponse;
+import com.google.cloud.translate.v3.TranslationServiceClient;
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+
+public class OcrTranslateText implements BackgroundFunction<PubSubMessage> {
+  // TODO<developer> set these environment variables
+  private static final String PROJECT_ID = System.getenv("GCP_PROJECT");
+  private static final String RESULTS_TOPIC_NAME = System.getenv("RESULT_TOPIC");
+
+  private static final Logger LOGGER = Logger.getLogger(OcrTranslateText.class.getName());
+  private static Publisher publisher;
+  private static final Gson gson = new Gson();
+  private static final String LOCATION_NAME = LocationName.of(PROJECT_ID, "global").toString();
+
+  public OcrTranslateText() throws IOException {
+    publisher = Publisher.newBuilder(
+        ProjectTopicName.of(PROJECT_ID, RESULTS_TOPIC_NAME)).build();
+  }
+
+  @Override
+  public void accept(PubSubMessage pubSubMessage, Context context) {
+    OcrTranslateApiMessage ocrMessage = OcrTranslateApiMessage.fromPubsubData(
+        pubSubMessage.getData().getBytes());
+
+    String targetLang = ocrMessage.getLang();
+    LOGGER.info("Translating text into " + targetLang);
+
+    // Translate text to target language
+    String text = ocrMessage.getText();
+    TranslateTextRequest request =
+        TranslateTextRequest.newBuilder()
+            .setParent(LOCATION_NAME.toString())
+            .setMimeType("text/plain")
+            .setTargetLanguageCode(targetLang)
+            .addContents(text)
+            .build();
+
+    TranslateTextResponse response;
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
+      response = client.translateText(request);
+    } catch (IOException e) {
+      // Cast to RuntimeException
+      throw new RuntimeException(e);
+    }
```

Comment:
This is still very wrong.

Reply:
Compiler error: `Overridden method accept(...) does not throw IOException`